server/handlers/shared_validators.js

```diff
@@ -23,3 +25,63 @@ exports.passwordBodyValidator = body("password")
   .isLength({ min: 6 })
   .trim()
   .escape();
+
+//products
+
+exports.productIdValidator = param("id").isMongoId();
+exports.farmerIdValidator = param("id").isMongoId();
+exports.productNameValidator = body("name")
+  .notEmpty()
+  .bail()
+  .isString()
+  .bail()
+  .isLength({ max: 25 })
+  .trim()
+  .escape();
+
+exports.productDescriptionValidator = body("description")
+  .notEmpty()
+  .bail()
+  .isString()
+  .bail()
+  .isLength({ max: 100 })
+  .trim()
+  .escape();
+
+exports.productCategoryValidator = body("category")
+  .if(body('category').exists())
+  .notEmpty()
+  .bail()
+  .isString()
+  .bail()
+  .isIn(Object.values(Product.Categories))
+  .trim()
```

Comment:
trim and escape are useless given the fact you restrict the possible values to just the categories names